### DISCUSSION TITLE
Why is Netdata now so difficult to install on CentOS7 and Apache?

### DISCUSSION THREAD

**Discussion by TheOldMan2000** [netdata/netdata#14229] [Q&A]
Hi all,

I'm sorry if this sounds a bit whiny, but after weeks of trying to get Netdata installed on my CentOS7 VPS which is a Virtuzzo container, latest WHM Cpanel. I've just about given up. 

It used to be fairly straightforward just using the one-line installer (aside from proxying it with Apache, which has never worked for me), but these days now all I get is a hellish mix of installs in /etc/netdata/etc/netdata, sometimes  /opt/netdata/etc/netdata, it fails to run under SystemD, Systemctl just says it stopped and failed. 

Then other times it runs but I can't ever connect to the 19999 dashboard at all, because the SSL doesn't work at all out of the box. I've never understood why it doesn't simply use or duplicate or convert the available certs upon install like other software does.

SSL used to work if I copied the root certs from WHM/Cpanel into /ssl/cert.pem and key.pem, then making the pens and folders readable to Netdata, but for weeks now all I get are errors in the browser; the key is too long, the cyphers don't overlap etc. It tells me I can use WGET or Curl to get the current conf, but I can't because there is no SSL!

Other times it won't create a working updater, or it can't find an uninstaller. 

It complains it can't find a compatible native build for CentOS 7. When it does half work it leaves half of itself behind and still running after uninstall!

It says to try a static build, try from GitHub source. The logs are installed in a weird location, I try to move them and they refuse after restarting even with temporary 0777 file and folder permissions and Netdata:Netdata permissions. I tried from GitHub source  but Cloud and updating doesn't work.

It still complains my OpenSSL isn't up to date, but it still looks in the wrong location for it and doesn't see the up to date version.

It's been a total nightmare to be honest!

I'd like it to be ask me where I want to install it, or smart enough to not install itself four directories deep! 

Also, I'd like it use my server hostname ssl certs out of the box.
I'd like a wizard to offer me where and how I want to access it, on the hostname, for each vhost, one to many by proxy etc. 

I'd like it to configure itself to set up a vhost, and just proxy without Nginx using my hostname or vhost Auto SSL certs.

That said, when it worked it was great, and I miss it. I've tried a number of software like Hetrix, Munin, NewRelic, they all can't hold a candle to Netdata, but I like how Newrelic tries to help you spot and diagnose issues with MySQL etc

**Comment by TheOldMan2000** [comment]
Another attempt!

```

wget -O /tmp/netdata-kickstart.sh https://my-netdata.io/kickstart.sh && sh /tmp/netdata-kickstart.sh
--2023-01-07 22:15:07--  https://my-netdata.io/kickstart.sh
Resolving my-netdata.io (my-netdata.io)... 172.67.156.192, 104.21.13.159, 2606:4700:3036::ac43:9cc0, ...
Connecting to my-netdata.io (my-netdata.io)|172.67.156.192|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/octet-stream]
Saving to: ‘/tmp/netdata-kickstart.sh’

    [ <=>                                                                    ] 81,037      --.-K/s   in 0.002s  

2023-01-07 22:15:07 (44.2 MB/s) - ‘/tmp/netdata-kickstart.sh’ saved [81037]


 --- Using /tmp/netdata-kickstart-cwW3lvJupz as a temporary directory. --- 
 --- Checking for existing installations of Netdata... --- 
 --- No existing installations of netdata found, assuming this is a fresh install. --- 
 --- Attempting to install using native packages... --- 
 --- Repository configuration is already present, attempting to install netdata. --- 
 --- Checking for libuv availability. --- 
[/tmp/netdata-kickstart-cwW3lvJupz]# env yum install netdata 
Loaded plugins: fastestmirror, tsflags, universal-hooks
Loading mirror speeds from cached hostfile
 * EA4: 63.247.64.58
 * cpanel-addons-production-feed: 63.247.64.58
 * cpanel-plugins: 63.247.64.58
 * epel: dl.fedoraproject.org
 * extras: us.mirror.nsec.pt
 * remi: mirror.pit.teraswitch.com
 * remi-php74: mirror.pit.teraswitch.com
 * remi-php80: mirror.pit.teraswitch.com
 * remi-php81: mirror.pit.teraswitch.com
 * remi-safe: mirror.pit.teraswitch.com
 * updates: coresite.mm.fcix.net
Resolving Dependencies
--> Running transaction check
---> Package netdata.x86_64 0:1.37.1-1.el7 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

=================================================================================================================
 Package                   Arch                     Version                         Repository              Size
=================================================================================================================
Installing:
 netdata                   x86_64                   1.37.1-1.el7                    epel                   3.1 M

Transaction Summary
=================================================================================================================
Install  1 Package

Total download size: 3.1 M
Installed size: 10 M
Is this ok [y/d/N]: y
Downloading packages:
netdata-1.37.1-1.el7.x86_64.rpm                                                           | 3.1 MB  00:00:00     
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Installing : netdata-1.37.1-1.el7.x86_64                                                                   1/1 
warning: group netdata does not exist - using root
warning: group netdata does not exist - using root
warning: group netdata does not exist - using root
warning: group netdata does not exist - using root
warning: group netdata does not exist - using root
warning: user netdata does not exist - using root
warning: group netdata does not exist - using root
warning: user netdata does not exist - using root
warning: group netdata does not exist - using root
warning: user netdata does not exist - using root
warning: group netdata does not exist - using root
sed: can't read /etc/netdata/netdata.conf: No such file or directory
The current config file can be downloaded with the following command
curl -o /etc/netdata/netdata.conf http://localhost:19999/netdata.conf
Config should be edited with /usr/libexec/netdata/edit-config
  Verifying  : netdata-1.37.1-1.el7.x86_64                                                                   1/1 

Installed:
  netdata.x86_64 0:1.37.1-1.el7                                                                                  

Complete!
 OK
``` 

 WARNING  Could not find netdata-updater.sh. This means that auto-updates cannot (currently) be enabled on this system. See https://learn.netdata.cloud/docs/agent/packaging/installer/update for more information about updating Netdata.

Successfully installed the Netdata Agent.

The following non-fatal warnings or errors were encountered:

  **- Could not find netdata-updater.sh. This means that auto-updates cannot (currently) be enabled on this system.** See https://learn.netdata.cloud/docs/agent/packaging/installer/update for more information about updating Netdata.
``

So I used the one line Kickstarter script, and it was as much use as a handbrake on a canoe.

According to the docs it points to at https://learn.netdata.cloud/docs/agent/packaging/installer/update , since v1.3.4 I can use:

> /usr/libexec/netdata/netdata-updater.sh --enable-auto-updates

> Response:
> -bash: /usr/libexec/netdata/netdata-updater.sh: No such file or directory

Great!

> root@vpsxxx [/]# locate -e netdata-updater
> root@vpsxxx [/]#_ 

Another failed install. It just leaves me hanging, twisting slowly in the wind.

**Comment by TheOldMan2000** [comment]
Next, the documentation says its time to do a reinstall.

> wget -O /tmp/netdata-kickstart.sh https://my-netdata.io/kickstart.sh && sh /tmp/netdata-kickstart.sh --reinstall

>  --- Using /tmp/netdata-kickstart-hUzAcado3Z as a temporary directory. --- 
 --- Checking for existing installations of Netdata... --- 
 WARNING  Found an existing netdata install managed by the system package manager, but could not determine the install type. Usually this means you installed an unsupported third-party netdata package.
__

So, now it's my fault? Why you little...

> ABORTED  Cancelling reinstallation at user request

Back to square one.

**Comment by TheOldMan2000** [comment]
Next, it says we can try a clean reinstall.

> wget -O /tmp/netdata-kickstart.sh https://my-netdata.io/kickstart.sh && sh /tmp/netdata-kickstart.sh --reinstall-clean

> --2023-01-07 22:49:03--  https://my-netdata.io/kickstart.sh
Resolving my-netdata.io (my-netdata.io)... 104.21.13.159, 172.67.156.192, 2606:4700:3036::ac43:9cc0, ...

> Connecting to my-netdata.io (my-netdata.io)|104.21.13.159|:443... connected.
> HTTP request sent, awaiting response... 200 OK
> Length: unspecified [application/octet-stream]
> Saving to: ‘/tmp/netdata-kickstart.sh’
> 
> 2023-01-07 22:49:04 (32.8 MB/s) - ‘/tmp/netdata-kickstart.sh’ saved [81037]
> 
> 
>  --- Using /tmp/netdata-kickstart-eaxPo1LkPp as a temporary directory. --- 
>  --- Checking for existing installations of Netdata... --- 
>  --- Downloading netdata-uninstaller ... --- 
> [/tmp/netdata-kickstart-eaxPo1LkPp]# curl --fail -q -sSL --connect-timeout 10 --retry 3 --output /tmp/netdata-kickstart-eaxPo1LkPp/netdata-uninstaller.sh https://raw.githubusercontent.com/netdata/netdata/master/packaging/installer/netdata-uninstaller.sh 
>  OK 
> 
> [/tmp/netdata-kickstart-eaxPo1LkPp]# /tmp/netdata-kickstart-eaxPo1LkPp/netdata-uninstaller.sh --yes 
> /tmp/netdata-kickstart-eaxPo1LkPp/netdata-uninstaller.sh: line 699: /etc/netdata/.environment: No such file or directory
> 
> Sat Jan  7 22:49:05 GMT 2023 : FATAL: netdata-uninstaller.sh: FAILED TO UNINSTALL NETDATA:  Failed to completely remove Netdata from this system.
>  **FAILED**
>  
> 
> >  --- Using /tmp/netdata-kickstart-KmkuiPSLGm as a temporary directory. --- 
>  --- Attempting to install using native packages... --- 
>  --- Repository configuration is already present, attempting to install netdata. --- 
>  --- Checking for libuv availability. --- 
> [/tmp/netdata-kickstart-KmkuiPSLGm]# env yum install netdata 
> Loaded plugins: fastestmirror, tsflags, universal-hooks
> Loading mirror speeds from cached hostfile
>  * EA4: 63.247.64.58
>  * cpanel-addons-production-feed: 63.247.64.58
>  * cpanel-plugins: 63.247.64.58
>  * epel: dl.fedoraproject.org
>  * extras: us.mirror.nsec.pt
>  * remi: mirror.pit.teraswitch.com
>  * remi-php74: mirror.pit.teraswitch.com
>  * remi-php80: mirror.pit.teraswitch.com
>  * remi-php81: mirror.pit.teraswitch.com
>  * remi-safe: mirror.pit.teraswitch.com
>  * updates: coresite.mm.fcix.net
> Package netdata-1.37.1-1.el7.x86_64 already installed and latest version
> Nothing to do
>  OK 
> 
>  **WARNING**  Could not find netdata-updater.sh. This means that auto-updates cannot (currently) be enabled on this system. See https://learn.netdata.cloud/docs/agent/packaging/installer/update for more information about updating Netdata.
>  
> Successfully installed the Netdata Agent.
> 
> The following non-fatal warnings or errors were encountered:
> 
>   - Command "/tmp/netdata-kickstart-eaxPo1LkPp/netdata-uninstaller.sh --yes" failed with exit code 1.
>   - Could not find netdata-updater.sh. This means that auto-updates cannot (currently) be enabled on this system. See https://learn.netdata.cloud/docs/agent/packaging/installer/update for more information about updating Netdata.
> 
> Official documentation can be found online at https://learn.netdata.cloud/docs/.
> 
>

**Comment by TheOldMan2000** [comment]
Well, it be a mangled install but it says it succeded despite providing no info on the conf or uninstaller.

`systemctl status -l netdata`

> ● netdata.service - Real time performance monitoring
>    Loaded: loaded (/usr/lib/systemd/system/netdata.service; disabled; vendor preset: disabled)
>    **Active: inactive (dead)**
> 
> Jan 07 17:53:18 vpsxxx.hostname.com systemd[1]: [/usr/lib/systemd/system/netdata.service:50] Failed to parse capability in bounding/ambient set, ignoring: CAP_PERFMON
> Jan 07 17:53:18 vpsxxx.hostname.com systemd[1]: [/usr/lib/systemd/system/netdata.service:69] Unknown lvalue 'ProtectControlGroups' in section 'Service'
> Jan 07 17:53:18 vpsxxx.hostname.com systemd[1]: [/usr/lib/systemd/system/netdata.service:50] Failed to parse capability in bounding/ambient set, ignoring: CAP_PERFMON
> Jan 07 17:53:18 vpsxxx.hostname.com systemd[1]: [/usr/lib/systemd/system/netdata.service:69] Unknown lvalue 'ProtectControlGroups' in section 'Service'
> Jan 07 22:15:25 vpsxxx.hostname.com systemd[1]: [/usr/lib/systemd/system/netdata.service:50] Failed to parse capability in bounding/ambient set, ignoring: CAP_PERFMON
> Jan 07 22:15:25 vpsxxx.hostname.com systemd[1]: [/usr/lib/systemd/system/netdata.service:69] Unknown lvalue 'ProtectControlGroups' in section 'Service'
> Jan 07 22:58:00 vpsxxx.hostname.com systemd[1]: [/usr/lib/systemd/system/netdata.service:50] Failed to parse capability in bounding/ambient set, ignoring: CAP_PERFMON
> Jan 07 22:58:00 vpsxxx.hostname.com systemd[1]: [/usr/lib/systemd/system/netdata.service:69] Unknown lvalue 'ProtectControlGroups' in section 'Service'
> Jan 07 22:58:34 vpsxxx.hostname.com systemd[1]: [/usr/lib/systemd/system/netdata.service:50] Failed to parse capability in bounding/ambient set, ignoring: CAP_PERFMON
> Jan 07 22:58:34 vpsxxx.hostname.com systemd[1]: [/usr/lib/systemd/system/netdata.service:69] Unknown lvalue 'ProtectControlGroups' in section 'Service'
>

**Comment by TheOldMan2000** [comment]
So, the docs say the only thing to do now is an uninstall... 

`wget -O /tmp/netdata-kickstart.sh https://my-netdata.io/kickstart.sh && sh /tmp/netdata-kickstart.sh --uninstall
--2023-01-07 23:04:49--  https://my-netdata.io/kickstart.sh
Resolving my-netdata.io (my-netdata.io)... 104.21.13.159, 172.67.156.192, 2606:4700:3031::6815:d9f, ...
Connecting to my-netdata.io (my-netdata.io)|104.21.13.159|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/octet-stream]
Saving to: ‘/tmp/netdata-kickstart.sh’

    [ <=>                                                                    ] 81,037      --.-K/s   in 0.004s  

2023-01-07 23:04:50 (18.8 MB/s) - ‘/tmp/netdata-kickstart.sh’ saved [81037]


 --- Using /tmp/netdata-kickstart-lB35ldXaOE as a temporary directory. --- 
 --- Checking for existing installations of Netdata... --- 
 --- Downloading netdata-uninstaller ... --- 
[/tmp/netdata-kickstart-lB35ldXaOE]# curl --fail -q -sSL --connect-timeout 10 --retry 3 --output /tmp/netdata-kickstart-lB35ldXaOE/netdata-uninstaller.sh https://raw.githubusercontent.com/netdata/netdata/master/packaging/installer/netdata-uninstaller.sh 
 OK 

[/tmp/netdata-kickstart-lB35ldXaOE]# /tmp/netdata-kickstart-lB35ldXaOE/netdata-uninstaller.sh --yes 
/tmp/netdata-kickstart-lB35ldXaOE/netdata-uninstaller.sh: line 699: /etc/netdata/.environment: No such file or directory

Sat Jan  7 23:04:50 GMT 2023 : FATAL: netdata-uninstaller.sh: FAILED TO UNINSTALL NETDATA:  Failed to completely remove Netdata from this system.
 FAILED 

Finished uninstalling the Netdata Agent.The following non-fatal warnings or errors were encountered:

  - Command "/tmp/netdata-kickstart-lB35ldXaOE/netdata-uninstaller.sh --yes" failed with exit code 1.`
  - 
  
But, that's broken too, so now full circle.

All I did was use the kickstart one liner, I didn't even opt out of the creepy analytics!

**Comment by TheOldMan2000** [comment]
Looks like I completely forgot about a previous support issue about these problems.  
https://github.com/netdata/netdata/issues/13213

I will try to use that to resolve.

**Comment by MrZammler** [comment]
Hi @TheOldMan2000 sorry to see you're having these issues with netdata. Clearly it's not our intent to put users under this stress :-)

Can you please provide your `/etc/os-release` output, so as to try and setup a similar VM to check the issues you're having? Thanks!

**Comment by TheOldMan2000** [comment]
I've finally got it working behind an existing host on Apache.

> NAME="CentOS Linux"
> VERSION="7 (Core)"
> ID="centos"
> ID_LIKE="rhel fedora"
> VERSION_ID="7"
> PRETTY_NAME="CentOS Linux 7 (Core)"
> ANSI_COLOR="0;31"
> CPE_NAME="cpe:/o:centos:centos:7"
> HOME_URL="https://www.centos.org/"
> BUG_REPORT_URL="https://bugs.centos.org/"
> CENTOS_MANTISBT_PROJECT="CentOS-7"
> CENTOS_MANTISBT_PROJECT_VERSION="7"
> REDHAT_SUPPORT_PRODUCT="centos"
> REDHAT_SUPPORT_PRODUCT_VERSION="7"

**Comment by TheOldMan2000** [comment]
I thought I'd got it working but today I realised it wasn't updating. After running the updater, I found out that the SystemD Netdata.service was incorrectly configured and I couldn't restart Netdata at all. 

Just constant errors:

> Unit netdata.service entered failed state.

and

> Job for netdata.service failed because a configured resource limit was exceeded. See "systemctl status netdata.service" and "journalctl -xe" for details.

By using:

`nano /usr/lib/systemd/system/netdata.service`

I found out these 2 lines inside should be:

`EnvironmentFile=/opt/netdata/etc/netdata/.environment`
`ExecStart=/opt/netdata/bin/netdata -D $EXTRA_OPTS`

They were both wrongly set to /etc/netdata/ and /usr/sbin/netdata both of which didn't exist on OS, because Netdata installed a static build in /opt/netdata.

After 1000's of nightly builds, it's still installing and configuring in varying directories, which causes nothing but problems and confusion.
